### DISCUSSION TITLE
Canvas Rendering Completion Detection: Method or Property Assessment

### DISCUSSION THREAD

**Discussion by RathodAnnu13** [fabricjs/fabric.js#9181] [Q&A]
Hello Everyone,

I would like to inquire whether Fabric.js provides a dedicated method or property that can reliably indicate when the canvas rendering process is completed. 

This feature would greatly assist in efficiently managing post-rendering actions within our application.

I am currently using Fabric version 1.7.22. In my current workflow, I am generating an image from the canvas in Javascript. The process involves downloading an SVG and rendering it onto the canvas. However, I am facing an issue where the image is being exported before the SVG is rendered onto the canvas. To address this issue, I attempted to utilize the **canvas.getObjects()** method to count the total number of objects on the canvas, but unfortunately, this approach did not yield the expected results. 
I would appreciate any guidance or insights on resolving this matter. 
Thank you.

**Comment by ShaMan123** [comment]
I did not understand your use case but the `render:after` hook indicates the rendering end, you can check the `ctx` param to undrestand what just finished the rendering

**Reply by ShaMan123** [reply]
Also you could simply override `renderAll` etc.

**Reply by RathodAnnu13** [reply]
please check below flowchart.
https://app.diagrams.net/#G1g_fk_bhk3A6xqnlkW8HJlHni71vzklC4
Here while arranging All canvas's object , image is exporting.
